tools/codegen/README.md

```diff
@@ -0,0 +1,53 @@
+# Weave SDK Code Generator
+
+This tool generates code from the OpenAPI specification for Python, Node.js, and TypeScript using Stainless.
+
+## Setup
+
+1. Install weave locally:
+
+   ```
+   uv pip install -e .
+   ```
+
+2. Set the required environment variables:
+
+   - `STAINLESS_API_KEY`
+   - `GITHUB_TOKEN`
+
+3. Configure your local paths:
+
+   ```bash
+   # Copy the template configuration file
+   cp tools/codegen/generate_config.yaml.template tools/codegen/generate_config.yaml
+
+   # Edit generate_config.yaml with your local repository paths
+   ```
+
+   Note: Your local `generate_config.yaml` will be ignored by git to prevent checking in personal paths.
+
+## Commands
+
+- **get-openapi-spec**: Starts a temporary server to fetch and save the OpenAPI spec.
```

Comment:
are these commands or "ARGS" values?

Reply:
They are both.  If you ran from  `tools/codegen` they would be commands, and if you run with `make` from repo root then they are ARGS (which are passed and become commands)

Reply:
I'll make a note about this as well